ArduCopter/mode_loiter.cpp

```diff
@@ -125,14 +125,28 @@ void ModeLoiter::run()
         pos_control->update_z_controller();
         break;
 
-    case AltHold_Takeoff:
+    case AltHold_Takeoff: {
+        // use rangefinder if healthy
+        const bool use_rngfnd = copter.rangefinder_alt_ok();
+
         // initiate take-off
         if (!takeoff.running()) {
-            takeoff.start(constrain_float(g.pilot_takeoff_alt,0.0f,1000.0f));
+            float alt_delta_cm = constrain_float(g.pilot_takeoff_alt, 0.0f, 1000.0f);
```

Comment:
Why hasn't this stuff been shoves into `start`?  You allude to the problem in your PR creation message - but don't explain why it's not in `start`...